dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -641,9 +641,24 @@ bool is_ip_equal(const ACE_INET_Addr& a, const DCPS::Locator_t& locator)
   return cmp_ip4(a, locator);
 }
 
+void print_locator(const CORBA::ULong i, const DCPS::Locator_t& o){
+  const unsigned char* a = reinterpret_cast<const unsigned char*>(o.address);
+  ACE_INET_Addr addr;
+  bool b = locator_to_address(addr, o, false) == 0;
+  ACE_DEBUG((LM_DEBUG, ACE_TEXT("locator%d[%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d.%d] locator_to_address:%C\n"),
```

Comment:
Will this work with ipv6?

Reply:
This prints debug message showing exactly what's in the locator.address no matter whether it's ip4, ip6 or other types. The function locator_to_address() is not used in the checks in this PR and just provides extra debug information here. If it failed, the message would indicate so.